grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -184,18 +184,18 @@ class ProductService {
 
 	void addStreamToProduct(Product product, Stream stream, SecUser currentUser)
 			throws ValidationException, NotPermittedException {
-		permissionService.verifyShare(currentUser, stream)
+		permissionService.verifyShareStream(currentUser, stream)
```

Comment:
We know that it's a `Stream`, so `permissionService.verify(currentUser, stream, Operation.STREAM_SHARE)` avoids the need for resource-specific methods in `PermissionService` (see earlier comment).